cmd/portal/cmd/cmdinternal/migrate_resources_set_default_logo_height.go

```diff
@@ -0,0 +1,180 @@
+	var darkThemeCSSMigrated []byte
+	var lightThemeCSSAlreadySet bool
+	var darkThemeCSSAlreadySet bool
+	if hasLightLogo {
+		r := bytes.NewReader(cfg.LightThemeCSS.DecodedData)
+		lightThemeCSSMigrated, lightThemeCSSAlreadySet, err = theme.MigrateSetDefaultLogoHeight(r)
+		if lightThemeCSSAlreadySet {
+			log.Printf("Skipping light theme css of app (%s) because it already has logo height set", appID)
+		}
+		if err != nil {
+			return fmt.Errorf("failed to migrate %v: %w", cfg.LightThemeCSS.OriginalPath, err)
+		}
+	}
+
+	if hasDarkLogo {
+		r := bytes.NewReader(cfg.DarkThemeCSS.DecodedData)
+		darkThemeCSSMigrated, darkThemeCSSAlreadySet, err = theme.MigrateSetDefaultLogoHeight(r)
+		if darkThemeCSSAlreadySet {
+			log.Printf("Skipping dark theme css of app (%s) because it already has logo height set", appID)
+		}
+		if err != nil {
+			return fmt.Errorf("failed to migrate %v: %w", cfg.DarkThemeCSS.OriginalPath, err)
+		}
+	}
+
+	if !lightThemeCSSAlreadySet {
+		configSourceData[cfg.LightThemeCSS.OriginalPath] = base64.StdEncoding.EncodeToString(lightThemeCSSMigrated)
+		if dryRun {
+			log.Printf("Before light-theme.css updated:")
+			log.Printf("%s:\n%s\n", cfg.LightThemeCSS.OriginalPath, string(cfg.LightThemeCSS.DecodedData))
+			log.Printf("After light-theme.css updated:")
+			log.Printf("%s:\n%s\n", cfg.LightThemeCSS.OriginalPath, string(lightThemeCSSMigrated))
+		}
+	}
+
+	if !darkThemeCSSAlreadySet {
+		configSourceData[cfg.DarkThemeCSS.OriginalPath] = base64.StdEncoding.EncodeToString(darkThemeCSSMigrated)
+		if dryRun {
+			log.Printf("Before dark-theme.css updated:")
+			log.Printf("%s:\n%s\n", cfg.DarkThemeCSS.OriginalPath, string(cfg.DarkThemeCSS.DecodedData))
+			log.Printf("After dark-theme.css updated:")
+			log.Printf("%s:\n%s\n", cfg.DarkThemeCSS.OriginalPath, string(darkThemeCSSMigrated))
+		}
+	}
+
+	return nil
+}
+
+type ResourceConfig struct {
+	OriginalPath string
+	EncodedData  string
+}
+
+type ResourceConfigDecoded struct {
+	OriginalPath string
+	EncodedData  string
+	DecodedData  []byte
+}
+type MigrateLogoHeightConfig struct {
+	LightLogo     *ResourceConfig
+	DarkLogo      *ResourceConfig
+	LightThemeCSS *ResourceConfigDecoded
+	DarkThemeCSS  *ResourceConfigDecoded
+}
+
+func parseLogoHeightConfigSource(configSourceData map[string]string) (*MigrateLogoHeightConfig, error) {
+	out := &MigrateLogoHeightConfig{}
+
+	for k, v := range configSourceData {
+		if strings.Contains(k, "logo") {
+			// found logo asset
+			if strings.Contains(k, "dark") {
+				out.DarkLogo = newResourceConfig(k, v)
+			} else {
+				out.LightLogo = newResourceConfig(k, v)
+			}
+		}
+		if strings.HasSuffix(k, "light-theme.css") {
+			rcd, err := newResourceConfigDecoded(k, v)
+			if err != nil {
+				return nil, err
+			}
+			out.LightThemeCSS = rcd
+		}
+		if strings.HasSuffix(k, "dark-theme.css") {
+			rcd, err := newResourceConfigDecoded(k, v)
+			if err != nil {
+				return nil, err
+			}
+			out.DarkThemeCSS = rcd
+		}
```

Comment:
The parsing here is not very robust, and can be broken in the future if we add a new resource that has similar names.

And this function does not have tests.

Reply:
> The parsing here is not very robust

Agreed, changed to absolute/regex path [in this commit later](https://github.com/authgear/authgear-server/pull/4776/commits/db5bf638af20dc9271b7dd2d39cace05ba68c404); sor for not fix-upping

> can be broken in the future

curious - I wrote this script with `I will only run this script once` in mind. What makes you think we will re-use this script in future? 🙏

Reply:
> I wrote this script with I will only run this script once in mind. What makes you think we will re-use this script in future?

We can re-use this script by copying this script, and make a new script. And the code style will be copied. This is what I want to prevent.